### ISSUE TITLE
Add TPU support

### ISSUE THREAD

**Issue by jacobbieker** [openclimatefix/graph_weather#27]
## Detailed Description
Currently, the graph neural network library dependencies don't support TPUs with pytorch geometric, or don't seem to at least because of custom kernels. We could add a Jax version for TPU support? The original model was implemented in Jax apparently.

## Context
Being able to use TPUs could speed up training quite a bit.

**Comment by vballoli** [comment]
Hey @jacobbieker, I wanted to start contributing to openclimatefix and take a shot at this if this feature is still of interest.

**Comment by aavashsubedi** [comment]
Is this still of interest? Thinking of working on it over a long time frame.  
The bottleneck would be getting the graph block working but I managed to get some of the MLP components functioning.
Using Flax(Linnen) + Jraph.

**Comment by aavashsubedi** [comment]
@jacobbieker

**Comment by jacobbieker** [comment]
Yeah, this is still of interest! It would be good to be able to use this on TPUs. Ideally still in PyTorch, although up for JAX as well.

**Comment by aavashsubedi** [comment]
Hi, 
Amazing! I wasn't aware that you can use XLA on PyTorch till just now! :  )
For inference I guess this would be the easiest thing: 

    import torch_xla.utils.serialization as xser
    
    model.load_state_dict(xser.load('model.pt'))
   
https://stackoverflow.com/questions/69328983/are-pytorch-trained-models-transferable-between-gpus-and-tpus
So there's less mismatch between versions, will stick to PyTorch (maybe there's a cool way to wrap TPU support around each GPU model will see! ). WIll read up on XLA : )

**Comment by CodeVishal-17** [comment]
Hi @jacobbieder, quick check-in on this issue.

@aavashsubedi — are you still planning to work on this? Totally understand if not.

If this is currently blocked or on hold, I’d be happy to help or take on a small, well-scoped subtask if that’s useful.
